src/utils/mod.rs

```diff
@@ -45,28 +41,32 @@ pub fn pad_bytes(address: Address) -> Vec<u8> {
 
 pub async fn wait_for_transaction(
     provider_l2: &JsonRpcClient<HttpTransport>,
-    transaction_hash: FieldElement,
+    transaction_hash: Felt,
     tag: &str,
 ) -> Result<(), anyhow::Error> {
     let transaction_receipt = get_transaction_receipt(provider_l2, transaction_hash).await;
 
     let transaction_status = transaction_receipt.ok().unwrap();
 
     match transaction_status {
-        Receipt(transaction_receipt) => {
-            log::trace!("txn : {:?} : {:?}", tag, transaction_receipt);
-            Ok(())
+        TransactionReceiptWithBlockInfo { receipt: TransactionReceipt::Invoke(receipt), .. } => {
+            log::trace!("txn : {:?} : {:?}", tag, receipt);
         }
-        PendingReceipt(..) => {
-            log::trace!("⏳ waiting for transaction : {:?}", transaction_hash);
-            sleep(Duration::from_secs(2)).await;
-            Box::pin(wait_for_transaction(provider_l2, transaction_hash, "")).await
+        TransactionReceiptWithBlockInfo { receipt: TransactionReceipt::DeployAccount(receipt), .. } => {
+            let contract_address = receipt.contract_address;
+            log::trace!("txn : {:?} : {:?}", tag, contract_address);
         }
-    }
+        _ => {
+            log::error!("Transaction status: {:?}", transaction_status);
+            panic!("Transaction failed");
+        }
+    };
+
+    Ok(())
 }
 
-pub fn convert_felt_to_u256(felt: StarkFelt) -> U256 {
-    U256::from_big_endian(felt.bytes())
+pub fn convert_felt_to_u256(felt: Felt) -> U256 {
+    U256::from_big_endian(&felt.to_bytes_le())
```

Comment:
If we use `from_big_endian` this should be `felt.to_bytes_be()`, no? Else we can use `U256::from_little_endian`

Reply:
resolved

Reply:
resolved